### ISSUE TITLE
pg authentication failure causes hang with 0.10.6

### ISSUE THREAD

**Issue by anewton1998** [reactiverse/reactive-pg-client#194]
I noticed an issue when upgrading from 0.10.5 to 0.10.6. Using Vertx, when connecting to a database with bad credentials, it just hangs in 0.10.6. In 0.10.5 I get an error as expected.

If you need a project to reproduce: https://github.com/SwanPipe/swanpipe_be.git (at commit 67ea62308d13fe01448fc9fdfed34193609fabec):

Put correct credentials in `src/main/resources/default-config.json` and then

```
vertx run -cp build/libs/swanpipe_be-0.0.3-SNAPSHOT.jar -conf src/main/resources/default-config.json com.swanpipe.verticles.Main
```
You may get some other error, which should be expected.

Now put incorrect credentials in and try again.

**Comment by vietj** [comment]
thanks for reporting this

**Comment by vietj** [comment]
it's indeed a regression

**Comment by vietj** [comment]
it is fixed in master, would you mind giving a try ?

**Comment by anewton1998** [comment]
Confirmed. Fixed in 0.10.7-SNAPSHOT
